### PR TITLE
Search: Use wp-admin page to customize settings

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Changes proposed in this Pull Request:
* Proof of concept. **Do not merge**!
* Adds a "Search Configuration" wp-admin page within the "Appearance" sidebar navigation group. This page can read/update Jetpack Search options. 
  Screen Shot 2021-05-18 at 3 37 32 PM

#### Jetpack product discussion
pcNPJE-7J-p2.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your site with a Jetpack Search subscription.
* Build the assets via `yarn build`.
* Ensure that menu item `Search` (in `Jetpack`) has been added to the wp-admin sidebar.
* Ensure that the `Jetpack Search` link opens a wp-admin page for configuring Jetpack Search.
* Ensure that the configuration panel for the `Jetpack Search` page works as expected.

### CODE REVIEW

**Comment by bluefuton** [comment]
@jsnmoon what do you think about consolidating all of the Search settings in this screen (including toggling on/off the search experience from Settings > Performance), so we don't need to send anyone off to Calypso?

File: projects/plugins/jetpack/_inc/client/search-dashboard/index.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+				label={ __( 'Overlay Trigger', 'jetpack' ) }
+				value={ trigger }
+				options={ [
+					{ label: __( 'Open when the user starts typing', 'jetpack' ), value: 'immediate' },
+					{ label: __( 'Open when results are available', 'jetpack' ), value: 'results' },
+				] }
+				onChange={ setTrigger }
+			/>
+			{ /* TODO: Fix the ColorPicker implementation */ }
+			{ /* <ColorPicker color={ color } onChangeComplete={ value => setColor( value.hex ) } /> */ }
```
[the remark this answers is not in the document]
Comment (jsnmoon) [reply projects/plugins/jetpack/_inc/client/search-dashboard/index.js]: Correct -- just pushed out a commit that uses a color type input for this PR.

**Comment by jsnmoon** [comment]
> what do you think about consolidating all of the Search settings in this screen (including toggling on/off the search experience from Settings > Performance), so we don't need to send anyone off to Calypso? (@bluefuton)

I think it makes a lot of sense to add that functionality to the wp-admin page. I'm not sure if we want to keep the control panel as a submenu within `Appearances` in this scenario, however. The `Settings` menu might be a better fit if we were to add the module/IS toggle to this page.

**Comment by bluefuton** [comment]
Closing this proof-of-concept in favour of @kangzj's PRs, starting with https://github.com/Automattic/jetpack/pull/20224.

**Pull request closed without merging**